common/common-ui/src/main/java/com/duckduckgo/common/ui/experiments/visual/store/VisualDesignExperimentDataStoreImpl.kt

```diff
@@ -16,149 +16,48 @@
 import com.duckduckgo.privacy.config.api.PrivacyConfigCallbackPlugin
 import com.squareup.anvil.annotations.ContributesBinding
 import com.squareup.anvil.annotations.ContributesMultibinding
+import dagger.SingleInstanceIn
 import javax.inject.Inject
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.combine
-import kotlinx.coroutines.flow.firstOrNull
-import kotlinx.coroutines.flow.flowOf
-import kotlinx.coroutines.flow.map
-import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
-import kotlinx.coroutines.runBlocking
 
 @ContributesBinding(
     scope = AppScope::class,
     boundType = VisualDesignExperimentDataStore::class,
 )
 @ContributesMultibinding(scope = AppScope::class, boundType = PrivacyConfigCallbackPlugin::class)
+@SingleInstanceIn(scope = AppScope::class)
 class VisualDesignExperimentDataStoreImpl @Inject constructor(
-    @VisualDesignExperiment private val store: DataStore<Preferences>,
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
-    experimentalUIThemingFeature: ExperimentalUIThemingFeature,
+    private val experimentalUIThemingFeature: ExperimentalUIThemingFeature,
 ) : VisualDesignExperimentDataStore, PrivacyConfigCallbackPlugin {
-    private companion object {
 
-        private const val KEY_EXPERIMENT_ENABLED = "KEY_EXPERIMENT_ENABLED"
-        private const val KEY_NAVIGATION_BAR_ENABLED = "KEY_NAVIGATION_BAR_ENABLED"
+    private val _experimentFeatureFlagEnabled = MutableStateFlow(experimentalUIThemingFeature.self().isEnabled())
 
-        private const val EXPERIMENT_ENABLED_BY_DEFAULT = false
-
-        private val NO_PARENT_FEATURE = flowOf(FeatureState(isAvailable = true, isEnabled = true))
-    }
-
-    private val _experimentState = ExperimentFeature(
-        store = store,
-        coroutineScope = appCoroutineScope,
-        targetToggle = experimentalUIThemingFeature.self(),
-        prefsKey = KEY_EXPERIMENT_ENABLED,
-        prefsDefault = EXPERIMENT_ENABLED_BY_DEFAULT,
-    )
-    override val experimentState: StateFlow<FeatureState> = _experimentState.state
-
-    private val _navigationBarState = ExperimentFeature(
-        store = store,
-        coroutineScope = appCoroutineScope,
-        parentStateFlow = experimentState,
-        targetToggle = experimentalUIThemingFeature.browserNavigationBar(),
-        prefsKey = KEY_NAVIGATION_BAR_ENABLED,
-        prefsDefault = true, // for now, navigation bar always enabled when the whole experiment enabled
-    )
-    override val navigationBarState: StateFlow<FeatureState> = _navigationBarState.state
-
-    init {
-        updateFeatureState()
-    }
+    override val isExperimentEnabled: StateFlow<Boolean> = _experimentFeatureFlagEnabled.asStateFlow()
```

Comment:
In this flow we will be able to combine other feature flags for competing experiments before computing the enabled state. cc @malmstein